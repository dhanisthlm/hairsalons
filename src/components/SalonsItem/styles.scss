.salons {
  .controls {
    display: flex;
    justify-content: space-between;
    padding: 10px;

    .heading {
      font-size: 18px;
    }

    .filter-icon {
      font-size: 20px;
      color: #D4AF37;
      transform: rotate(-90deg);
    }

    .go-back {
      font-size: 18px;
      color: #D4AF37;
    }
  }

  .price-budget {
    height: 20px;
    display: flex;
    justify-content: space-between;
    padding: 10px 15px;
    border-color: #D4AF37;
    border-width: 1px 0 1px 0;
    border-style: solid;

    .price-filter-indicator {
      color: #D4AF37;
    }

    .toggle {
      transform: rotate(-90deg);
      color: #D4AF37;
    }
  }

  .list {
    margin: 0;
    padding: 15px;
    list-style-type: none;

    &.filter {
      margin-top: 105px;
    }

    .list-item {
      padding: 10px 0;
      border-bottom: 1px solid #EFEFEF;
      opacity: 0.7;
      cursor: pointer;

      .content {
        position: relative;
        padding-right: 27px;

        &:after {
          content: '\232A';
          position: absolute;
          top: 24px;
          right: -9px;
          color: #D4AF37;
        }

        .rating {
          overflow: hidden;
          padding-left: 50px;

          .stars {
            float: left;
            padding: 5px 0;

            .star {
              position: relative;
              color: #D4AF37;
              margin-right: 2px;
              font-size: 12px;

              &.filled:before {
                content: "\2605";
                position: absolute;
                top: -3px;
              }
            }

            .votes {
              font-family: 'Helvetica', 'Arial', 'sans-serif';
              margin-left: 5px;
              font-size: 12px;
            }
          }
        }
      }

      .address {
        font-family: 'Helvetica', 'Arial', 'sans-serif';
        font-size: 14px;
        font-weight: 300;
        display: block;
        margin-left: 49px;
        margin-top: 2px;
      }

      .duration {
        float: right;
        font-family: 'Helvetica', 'Arial', 'sans-serif';
        font-size: 14px;
        font-weight: 300;
        line-height: 27px;
      }

      .title {
        overflow: hidden;

        .time-slot {
          float: left;
          margin-right: 15px;
        }

        .price {
          float: right;
        }

        .name {
          font-size: 18px;
        }

        .price, .time-slot {
          font-family: 'Helvetica', 'Arial', 'sans-serif';
          font-size: 14px;
          line-height: 23px;
          font-weight: 500;
        }
      }
    }
  }
}