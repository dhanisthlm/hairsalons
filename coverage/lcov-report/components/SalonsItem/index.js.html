<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/SalonsItem/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/SalonsItem</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>2/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>2/3</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.scss';
&nbsp;
export const SalonsItem = <span class="fstat-no" title="function not covered" >({</span> item, handleClick }) =&gt; (
<span class="cstat-no" title="statement not covered" >    &lt;li</span>
        id={item._id}
        className="list-item"
        key={item._id}
        onClick={handleClick}&gt;
        {console.log(item)}
        &lt;div className="content"&gt;
            &lt;div className="title"&gt;
                &lt;span className="time-slot"&gt;{item.timeSlot}&lt;/span&gt;
                &lt;span className="name"&gt;{item.name}&lt;/span&gt;
                &lt;span className="price"&gt;{item.price} kr&lt;/span&gt;
            &lt;/div&gt;
            &lt;div className="rating"&gt;
                &lt;div className="stars"&gt;
                &lt;span className="star filled"&gt;☆&lt;/span&gt;
                &lt;span className="star filled"&gt;☆&lt;/span&gt;
                &lt;span className="star filled"&gt;☆&lt;/span&gt;
                &lt;span className="star filled"&gt;☆&lt;/span&gt;
                &lt;span className="star"&gt;☆&lt;/span&gt;
                &lt;span className="votes"&gt;(24)&lt;/span&gt;
            &lt;/div&gt;
            &lt;span className="duration"&gt;{item.duration} min&lt;/span&gt;
        &lt;/div&gt;
        &lt;span className="address"&gt;{item.address}&lt;/span&gt;
    &lt;/div&gt;
  &lt;/li&gt;
);
&nbsp;
SalonsItem.propTypes = {
    item: PropTypes.object.isRequired,
    handleClick: PropTypes.func.isRequired
};</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Dec 31 2018 00:04:19 GMT+0100 (centraleuropeisk normaltid)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
