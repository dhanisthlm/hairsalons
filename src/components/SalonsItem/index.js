import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.scss';

export const SalonsItem = ({ item, handleClick }) => (
    <li
        id={item._id}
        className="list-item"
        key={item._id}
        onClick={handleClick}>
        {console.log(item)}
        <div className="content">
            <div className="title">
                <span className="time-slot">{item.timeSlot}</span>
                <span className="name">{item.name}</span>
                <span className="price">{item.price} kr</span>
            </div>
            <div className="rating">
                <div className="stars">
                <span className="star filled">☆</span>
                <span className="star filled">☆</span>
                <span className="star filled">☆</span>
                <span className="star filled">☆</span>
                <span className="star">☆</span>
                <span className="votes">(24)</span>
            </div>
            <span className="duration">{item.duration} min</span>
        </div>
        <span className="address">{item.address}</span>
    </div>
  </li>
);

SalonsItem.propTypes = {
    item: PropTypes.object.isRequired,
    handleClick: PropTypes.func.isRequired
};