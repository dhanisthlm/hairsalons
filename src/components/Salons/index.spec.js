import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import configureStore from 'redux-mock-store'
import SalonList from './index';
import { SalonsItem } from '../SalonsItem';

Enzyme.configure({ adapter: new Adapter() })

const mockStore = configureStore();

const initialState = {
  salons: {
    list: [
      {
        name: 'Klippt och skuret',
        address: 'Kungsgatan 4',
        description: 'Stans bÃ¤sta salong'
      }
    ]
  }
};

let store, wrapper;

beforeEach(()=>{
  store = mockStore(initialState)
})

test('salonsitem', () => {
  wrapper = shallow(<SalonList store={store} />);
  expect(SalonsItem.length).toBe(2);
})