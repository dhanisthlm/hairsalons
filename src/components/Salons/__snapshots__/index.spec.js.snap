// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(SalonList)
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "dispatch": [Function],
      "filteredSalons": undefined,
      "priceRange": undefined,
      "salon": undefined,
      "salons": Array [
        Object {
          "address": "foo",
          "description": "foo",
          "name": "foo",
        },
      ],
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "dispatch": [Function],
        "filteredSalons": undefined,
        "priceRange": undefined,
        "salon": undefined,
        "salons": Array [
          Object {
            "address": "foo",
            "description": "foo",
            "name": "foo",
          },
        ],
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
